packages/node_modules/@webex/plugin-devicemanager/src/devicemanager.js

```diff
@@ -0,0 +1,503 @@
+      api: 'wdm',
+      resource: 'devices/auxiliary'
+    })
+      .then((res) => {
+        if (!res.body) {
+          return Promise.reject();
+        }
+        return Promise.all(res.body.items.map((device) => {
+          if (device.deviceClass === UC_CLOUD) {
+            device.id = `${this.spark.internal.device.services.wdmServiceUrl}/${device.id}`;
+          }
+          return this._decryptDeviceName(device);
+        }))
+          .then(() => {
+            /* eslint-disable consistent-return */
+            res.body.items.forEach((device) => {
+              if (device.deviceInfo
+                && device.deviceInfo.machineType === LYRA_SPACE) {
+                return this.spark.internal.lyra.space.get(device.deviceInfo)
+                  .catch((err) => {
+                    this.logger.error('DeviceManager#refresh: failed to receive device info', err);
+                  });
+              }
+            });
+            /* eslint-enable consistent-return */
+            res.body.items.forEach((device) => {
+              DeviceCollection.set(device);
+            });
+            return this.list();
+          });
+      })
+      .catch((err) => {
+        this.logger.error('DeviceManager#refresh: failed to fetch recent devices', err);
+      });
+  },
+
+  /**
+  * Search for a device by name
+  * @param {Object} options
+  * @param {string} options.searchQuery
+  * @returns {Promise<Device>}
+  */
+  search(options) {
+    if (!options || !options.searchQuery) {
+      this.logger.error('DeviceManager#search: options.searchQuery is required');
+      return Promise.reject(new Error('DeviceManager#search: options.searchQuery is required'));
+    }
+    return this.spark.internal.search.people({
+      searchId: uuid.v4(),
+      searchType: 'DEVICE_SEARCH',
+      searchEntity: 'device',
+      includePeople: false,
+      includeRooms: true,
+      queryString: options.searchQuery
+    })
+      .catch((err) => {
+        this.logger.error('DeviceManager#search: failed to search a device', err);
+      });
+  },
+
+  /**
+  * Caches the device info and also registers to Redis for subsequent fetches
+  * @param {Object} device
+  * @param {string} device.id
+  * @returns {deviceInfo}
+  */
+  upsert(device) {
+    const deviceId = device.id || device.identity && device.identity.id;
+    if (!deviceId) {
+      this.logger.error('DeviceManager#upsert: device.id is required');
+      return Promise.reject(new Error('DeviceManager#upsert: device.id is required'));
+    }
+    this._pairedDevice = this._devicePendingPinChallenge;
+    this._devicePendingPinChallenge = null;
+    // check if the device is already existing, if so then merge else add
+    const existingDevice = DeviceCollection.get(deviceId);
+    if (existingDevice) {
+      DeviceCollection.set(device);
+      return Promise.resolve();
+    }
+    // new device requested, add to wdm for subsequent retreivals
+    return this.spark.request({
```

Comment:
I checked there code, it was logical to break the requests out in the meeting's code since the same code was being called in several places unlike here where the code belongs to the current function, still I am open to suggestions and can make changes based on feedback.